src/compiler/crystal/compiler.cr

```diff
@@ -750,7 +789,7 @@ module Crystal
       end
     {% end %}
 
-    protected def optimize(llvm_mod)
+    protected def optimize(llvm_mod, target_machine)
```

Comment:
I'm not sure how the legacy pass manager actually fares. We share an instance, so it's likely a bad idea... maybe each thread should have its own pass manager/manager builder? or maybe even create a new one on each compilation unit (as we do with the pass builder) :thinking:

Reply:
**Confirmed**: I built a compiler with LLVM 12 and MT enabled (plus --single-module and -O1), then tried to recompile the compiler with -O1 on an empty cache, so we get 4 threads compiling and optimizing in parallel the 1905 objects, and I almost immediately got a segfault.